repose-aggregator/components/filters/saml-policy-translation-filter/src/main/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilter.scala

```diff
@@ -97,13 +100,25 @@ class SamlPolicyTranslationFilter @Inject()(configurationService: ConfigurationS
   }
 
   /**
-    * Gets the SAMl response from the post encoded body, and decodes it into a plain String.
+    * Gets the SAMl response from the post encoded body, and decodes it.
     *
     * @param request the servlet request that has an encoded body
     * @return the decoded saml response
     * @throws SamlPolicyException if decoding fails
     */
-  def decodeSamlResponse(request: HttpServletRequest): InputStream = ???
+  def decodeSamlResponse(request: HttpServletRequest): InputStream = {
+    try {
+      Option(request.getParameter("SAMLResponse"))
+        .map(Base64.getDecoder.decode)
```

Comment:
Also interesting, Java's `Base64` decoder that I get here uses `ISO-8859-1` to decode when passed a `String`. That works just fine for us, because `US-ASCII` is a subset of `ISO-8859-1`.